src/providers/DevicesProvider/index.tsx

```diff
@@ -2,13 +2,21 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import React from 'react';
+import { Device, AudioTransformDevice } from 'amazon-chime-sdk-js';
 
 import { AudioInputProvider, useAudioInputs } from './AudioInputProvider';
 import { AudioOutputProvider, useAudioOutputs } from './AudioOutputProvider';
 import { useVideoInputs, VideoInputProvider } from './VideoInputProvider';
 
-const DevicesProvider: React.FC = ({ children }) => (
-  <AudioInputProvider>
+interface Props {
+  onDeviceReplacement?: (
+    nextDevice: string,
+    currentDevice: Device | AudioTransformDevice
+  ) => Promise<Device | AudioTransformDevice>;
+}
+
+const DevicesProvider: React.FC<Props> = ({ children, onDeviceReplacement }) => (
+  <AudioInputProvider onDeviceReplacement={onDeviceReplacement}>
```

Comment:
IIRC React lets you write `{onDeviceReplacement}` instead.

Reply:
Thanks for the advice, seems like I need to write like this
```
<AudioInputProvider {...{onDeviceReplacement}}>
```
So just keep it as is. Please let me know if you are fine with it

Reply:
I was thinking of Svelte. As-is is fine. Thanks for checking!